pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/queries/ReportDeadLetterQueryService.kt

```diff
@@ -0,0 +1,38 @@
+package gov.cdc.ocio.processingstatusapi.queries
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.server.operations.Query
+import gov.cdc.ocio.processingstatusapi.dataloaders.ReportDeadLetterDataLoader
+import gov.cdc.ocio.processingstatusapi.loaders.ReportDeadLetterLoader
+import gov.cdc.ocio.processingstatusapi.models.ReportDeadLetter
+import graphql.schema.DataFetchingEnvironment
+import java.util.concurrent.CompletableFuture
+
+class ReportDeadLetterQueryService : Query {
+
+    @GraphQLDescription("Return all the dead-letter reports associated with the provided uploadId")
+    @Suppress("unused")
+    fun getDeadLetterReportsByUploadId(uploadId: String) = ReportDeadLetterLoader().getByUploadId(uploadId)
+
+    @GraphQLDescription("Return all the dead-letter reports associated with the provided datastreamId, datastreamroute and timestamp date range")
+    @Suppress("unused")
+    fun getDeadLetterReportsByDataStream(dataStreamId: String, dataStreamRoute:String, startDate:String, endDate:String)
```

Comment:
Please allow for either start/end date OR days interval.  We have a pattern for this in most of our queries to where you can either supply a start date and optional end date or simply the number of days.  You should be able to find lots of examples for this.

Reply:
Done. Please review